src/cart.ts

```diff
@@ -0,0 +1,25 @@
+import { CartItem } from './cart.types';
+
+export class ShoppingCart {
+    private items: CartItem[] = [];
+
+    public addItem(item: CartItem): void {
+        this.items.push(item);
+    }
+
+    public calculateTotal(discountPercentage: number = 0, taxRate: number = 0.1): number {
+        let total = 0;
+
+        for (const item of this.items) {
+            total += item.price * item.quantity;
+        }
+
+        if (discountPercentage > 0) {
+            total = total - discountPercentage;
```

Comment:
Hier gehe ich davon aus die calculation nicht korrekt ist. Wenn ich das richtig denke müsste es so gehen weil wir nicht den prozentwert selbst vom total abziehen sonder den ausgerechneten Rabat.

Den Teil könnte man auch noch sinnvoll in eine function auslagern (calculateDiscount):
(discountPercentage / 100) * total

Mit der function könnte man dann einfach:
total - calculateDiscount()